### ISSUE TITLE
Break loop using Factorization() on permutation groups

### ISSUE THREAD

**Issue by grahamee** [gap-system/gap#679]
This works OK:
```
gap> G:=Group((1,2,3,4));
Group([ (1,2,3,4) ])
gap> Factorization(G,());
<identity ...>
gap> Factorization(G,Elements(G)[1]);
<identity ...>
```
But if I start again and do these in a different order I get errors:
```
gap> G:=Group((1,2,3,4));
Group([ (1,2,3,4) ])
gap> Factorization(G,Elements(G)[1]);
Error, Record: '<rec>.orbitpos' must have an assigned value in
  pos := pos + val * S.orbitpos[img]; at /proc/cygdrive/C/gap4r8/lib/grp.gi:4779 called from
numobj( objelm( one ) ) at /proc/cygdrive/C/gap4r8/lib/grp.gi:4819 called from
<function "GenericFactorizationGroup">( <arguments> )
 called from read-eval loop at line 14 of *stdin*
you can 'return;' after assigning a value
brk> quit;
gap> Factorization(G,());
Error, Record: '<rec>.orbitpos' must have an assigned value in
  pos := pos + val * S.orbitpos[img]; at /proc/cygdrive/C/gap4r8/lib/grp.gi:4779 called from
numobj( objelm( one ) ) at /proc/cygdrive/C/gap4r8/lib/grp.gi:4819 called from
<function "GenericFactorizationGroup">( <arguments> )
 called from read-eval loop at line 14 of *stdin*
you can 'return;' after assigning a value
brk> quit;
gap>
```
Once this error happens, no call to Factorization() for G will succeed until we reset G again:
```
gap> G:=Group((1,2,3,4));
Group([ (1,2,3,4) ])
gap> Factorization(G,());
<identity ...>
```
This is on 4.8.2 (Windows and Linux) but appears to be present at least as far back as 4.7.2. It only seems to affect permutation groups, and not all - for example Group((1,2,3)) does not show this behaviour.

**Comment by ChrisJefferson** [comment]
This is indeed a bug -- `orbitpos` (and later `ol`) is only added in some situations, but this code is assuming it will always be present.

There are two obvious fixes -- make sure `orbitpos` and `ol` get set up (extracting their definitions out into a method), or just remove them.

**Comment by hulpke** [comment]
I have corrected this in my `fixes' branch.

**Comment by hulpke** [comment]
Corrected in #682
